src/main/assets/js/postcodes.ts

```diff
@@ -0,0 +1,126 @@
+import $ from 'jquery';
+import {AjaxErrorHandler} from './ajaxErrorHandler';
+import {Utilities} from './utilities';
+
+const { initAll } = require('govuk-frontend');
+
+export class PostcodesController {
+
+  private tabId = '#postcodesTab';
+  private postcodesNavTab = '#tab_postcodes'
+  private postcodesContentId = '#postcodesContent';
+  private addPostcodesBtnClass = 'addPostcodes';
+  private movePostcodesBtnClass = 'movePostcodesButton';
+  private selectAllPostcodes = 'postcodesSelectAllItems';
+  private deletePostcodesBtnClass = 'deletePostcodes';
+
+  constructor() {
+    this.initialize();
+  }
+
+  private initialize(): void {
+    $(() => {
+      if ($(this.tabId).length > 0) {
+        Utilities.toggleTabEnabled(this.postcodesNavTab, false);
+        this.getPostcodes();
+        this.setUpAddEventHandler();
+        this.setUpSelectAllEventHandler();
+        this.setUpDeleteEventHandler();
+        this.setUpMoveEventHandler();
+      }
+    });
+  }
+
+  private getPostcodes(): void {
+    const slug = $('#slug').val();
+
+    $.ajax({
+      url: `/courts/${slug}/postcodes`,
+      method: 'get',
+      success: (res) => {
+        this.updateContent(res);
+      },
+      error: (jqxhr) =>
+        AjaxErrorHandler.handleError(jqxhr, 'GET postcodes failed.')
+    });
+  }
+
+  private setUpAddEventHandler(): void {
+    $(this.tabId).on('click', `button.${this.addPostcodesBtnClass}`, e => {
+      e.preventDefault();
+      const slug = $('#slug').val();
+      const requestBodyData = PostcodesController.getRequestBodyData();
+      requestBodyData['newPostcodes'] = $('#addNewPostcodes').val();
+
+      $.ajax({
+        url: `/courts/${slug}/postcodes`,
+        method: 'post',
+        data: requestBodyData
+      }).done(res => {
+        this.updateContent(res);
+        window.scrollTo(0, 0);
+      }).fail(response =>
+        AjaxErrorHandler.handleError(response, 'POST new postcodes failed.'));
+    });
+  }
+
+  private setUpMoveEventHandler(): void {
+    $(this.tabId).on('click', `button.${this.movePostcodesBtnClass}`, e => {
+      e.preventDefault();
+      const slug = $('#slug').val();
+      const requestBodyData = PostcodesController.getRequestBodyData();
+      requestBodyData['selectedCourt'] = $('#movePostcodesSelect').val();
```

Comment:
You have some id's as variables at the start of the class and some like this. Might be better keeping it consistent.